libsplinter/src/admin/store/mod.rs

```diff
@@ -0,0 +1,454 @@
+pub enum RouteType {
+    Any,
+}
+
+impl Default for RouteType {
+    fn default() -> Self {
+        RouteType::Any
+    }
+}
+
+/// Native representation of a node included in circuit
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct CircuitNode {
+    id: String,
+    endpoints: Vec<String>,
+}
+
+/// Native representation of a node in a proposed circuit
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct ProposedNode {
+    node_id: String,
+    endpoints: Vec<String>,
+}
+
+/// Native representation of a service that is a part of circuit
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct Service {
+    service_id: String,
+    service_type: String,
+    allowed_nodes: Vec<String>,
+    arguments: Vec<(String, String)>,
+}
+
+/// Native representation of a service that is a part of a proposed circuit
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct ProposedService {
+    service_id: String,
+    service_type: String,
+    allowed_nodes: Vec<String>,
+    arguments: Vec<(String, String)>,
+}
+
+/// The unique ID of service made up of a circuit ID and the individual service ID.
+/// A service ID is only required to be unique from within a circuit.
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct ServiceId {
+    circuit_id: String,
+    service_id: String,
+}
+
+impl ServiceId {
+    /// Create a new service ID
+    ///
+    /// # Arguments
+    ///
+    /// * `circuit_id` - the ID of the circuit the service belongs to
+    /// * `service_id` - the individual ID for the service
+    pub fn new(circuit_id: String, service_id: String) -> Self {
+        ServiceId {
+            circuit_id,
+            service_id,
+        }
+    }
+
+    /// Returns the circuit ID
+    pub fn circuit(&self) -> &str {
+        &self.circuit_id
+    }
+
+    /// Returns the service ID
+    pub fn service_id(&self) -> &str {
+        &self.service_id
+    }
+
+    /// Decompose this ServiceId into a tuple of (<circuit ID>, <service ID>).
+    pub fn into_parts(self) -> (String, String) {
+        (self.circuit_id, self.service_id)
+    }
+}
+
+impl fmt::Display for ServiceId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}::{}", self.circuit_id, self.service_id)
+    }
+}
+
+impl Eq for ServiceId {}
+
+impl Ord for ServiceId {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let compare = self.circuit_id.cmp(&other.circuit_id);
+        if compare == Ordering::Equal {
+            self.service_id.cmp(&other.service_id)
+        } else {
+            compare
+        }
+    }
+}
+
+impl PartialOrd for ServiceId {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+/// Predicate for filtering the lists of circuits and circuit proposals
+pub enum CircuitPredicate {
+    ManagmentType(String),
+    Members(Vec<String>),
```

Comment:
We should probably change these variants to be more specific: `ManagementTypeEq` and `MembersIncludes`. This is more clear, and it gives us flexibility to add more variants in the future like `MembersExcludes` or `ManagementTypeNe`.